cirq/ops/parallel_gate_operation_test.py

```diff
@@ -0,0 +1,176 @@
+    assert cirq.is_parameterized(op1)
+    op2 = cirq.resolve_parameters(op1, r)
+    assert not cirq.is_parameterized(op2)
+
+
+def test_unitary():
+    a = cirq.NamedQubit('a')
+    b = cirq.NamedQubit('b')
+    g = cirq.SingleQubitGate()
+    p = cirq.ParallelGateOperation(g, [a,b])
+    q = cirq.ParallelGateOperation(cirq.X, [a,b])
+
+    assert not cirq.has_unitary(p)
+    assert cirq.unitary(p, None) is None
+    np.testing.assert_allclose(cirq.unitary(q),
+                               np.array([
+                                   [0. + 0.j, 0. + 0.j, 0. + 0.j, 0. + 0.j],
+                                   [0. + 0.j, 1. + 0.j, 1. + 0.j, 0. + 0.j],
+                                   [0. + 0.j, 1. + 0.j, 1. + 0.j, 0. + 0.j],
+                                   [0. + 0.j, 0. + 0.j, 0. + 0.j, 0. + 0.j]
+                               ]),
+                               atol=1e-8)
+
+
+def test_not_implemented_diagram():
+    a = cirq.NamedQubit('a')
+    g = cirq.SingleQubitGate()
+    c = cirq.Circuit()
+    c.append(cirq.ParallelGateOperation(g,[a]))
+    assert 'cirq.ops.gate_features.SingleQubitGate' in str(c)
+
+
+def test_repr():
+    a, b = cirq.LineQubit.range(2)
+    assert repr(cirq.ParallelGateOperation(cirq.X, (a, b))
+                ) == 'cirq.ParallelGateOperation(gate=cirq.X,' \
+                     ' qubits=[cirq.LineQubit(0), cirq.LineQubit(1)])'
+
+
+def test_str():
+    a, b = cirq.LineQubit.range(2)
+    assert str(cirq.ParallelGateOperation(cirq.X, (a,b))) == 'X(0, 1)'
+
+
+def test_phase():
+    g1 = cirq.SingleQubitGate()
+    g2 = cirq.S
+    g3 = cirq.phase_by(g2, 1, 0)
+    qreg = cirq.LineQubit.range(2)
+    op1 = cirq.ParallelGateOperation(g1, qreg)
+    op2 = cirq.ParallelGateOperation(g2, qreg)
+    assert cirq.phase_by(op2, 1, 0) == cirq.ParallelGateOperation(g3, qreg)
+    with pytest.raises(TypeError):
+        cirq.phase_by(op1, 1, 0)
+
+
+def test_equivalent_circuit():
+    depth = 6
+    qreg = cirq.LineQubit.range(4)
+    oldc = cirq.Circuit()
+    newc = cirq.Circuit()
+    exponents = [0, -1, 1, 1/2, 1/3, 1/3+0.01]
+
+    for moment in range(depth):
+        if moment == 0:
+            gate = cirq.XPowGate(exponent=exponents[moment])
+        if moment == 1:
+            gate = cirq.T**-1
+        if moment == 2:
+            gate = cirq.ZPowGate(exponent=exponents[moment])
+        if moment == 3:
+            gate = cirq.XPowGate(exponent=exponents[moment])
+        if moment == 4:
+            gate = cirq.YPowGate(exponent=exponents[moment])
+        if moment == 5:
+            gate = cirq.ZPowGate(exponent=exponents[moment])
+
+        for qubit in qreg:
+            oldc.append(gate.on(qubit))
+
+        newc.append(cirq.ops.ParallelGateOperation(gate, qreg))
+
+    assert str(oldc) == str(newc)
+    sim = cirq.Simulator()
+    old_result = sim.simulate(oldc)
+    new_result = sim.simulate(newc)
+    assert old_result.dirac_notation() == new_result.dirac_notation()
```

Comment:
Use `cirq.testing.assert_circuits_with_terminal_measurements_are_equivalent`

(your circuit is vacuously one with only terminal measurements)

Reply:
Done